doc/cpp_recipe.rst

```diff
@@ -0,0 +1,230 @@
+Recipes
+===============
+
+An Arbor recipe is a description of a model without any information about how the
+model is to be executed. It  provides interface that the model building phase queries
+for information on cells, such as:
+
+  * the number of cells in the model;
+  * the type of a cell;
+  * a description of a cell;
+  * incoming network connections on a cell.
+
+The :cpp:class:`arb::recipe` class documentation is below.
+
+Why Recipes?
+--------------
+
+The Recipe design was motivated by the following aims:
+
+    * Building a simulation from a recipe description must be possible in a
+      distributed system efficiently with minimal communication.
+    * The model description (i.e. recipe) should be seperate from the
+      instantiation of the model to execute on the target system.
+    * To minimise the amount of memory used in model building, to make it
+      possible to build and run simulations in one run.
+
+Recipe descriptions are cell-oriented, in order that the building phase can
+distributed and that the recipe description can be built indepedently of any
+runtime execution environment.
```

Comment:
It's not so much that the recipe itself can be built independently of a runtime environment, but that the semantics of the recipe are independent of the nature of the distribution or runtime environment.